src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/documents/DocumentApiClientTest.kt

```diff
@@ -49,34 +49,47 @@ class DocumentApiClientTest {
 
   @Test
   fun `Returns the document response when successfully uploading a document to the remote service`() {
-    stubUploadDocumentEndpoint(status = 201, responseBody = happyResponse)
+    givenDocumentApiRespondsWith(status = 201, responseBody = happyResponse)
 
-    val result = documentApiClient.uploadDocument(
-      file = file.contentAsByteArray,
-      documentUuid = UUID.fromString("e2487a03-7cf9-4a9c-85e4-1d51efd7b3f1"),
-      documentType = DocumentType.EXCLUSION_ZONE_MAP,
-      metadata = mapOf("aKey" to "1", "bKey" to "2", "cKey" to "3"),
-    )
+    val result = uploadDocument()
 
-    assertEquals(result?.documentUuid, UUID.fromString("e2487a03-7cf9-4a9c-85e4-1d51efd7b3f1"))
-    assertEquals(result?.documentType, DocumentType.EXCLUSION_ZONE_MAP)
+    assertEquals(result.documentUuid, UUID.fromString("e2487a03-7cf9-4a9c-85e4-1d51efd7b3f1"))
+    assertEquals(result.documentType, DocumentType.EXCLUSION_ZONE_MAP)
   }
 
   @ParameterizedTest
   @CsvSource("400", "401", "403", "409", "500")
   fun `Throws an exception when the request is not successful`(responseStatusCode: Int) {
-    stubUploadDocumentEndpoint(status = responseStatusCode, responseBody = errorResponse)
-
-    assertThrows<IllegalStateException> {
-      documentApiClient.uploadDocument(
-        file = file.contentAsByteArray,
-        documentUuid = UUID.fromString("e2487a03-7cf9-4a9c-85e4-1d51efd7b3f1"),
-        documentType = DocumentType.EXCLUSION_ZONE_MAP,
-        metadata = mapOf("aKey" to "1", "bKey" to "2", "cKey" to "3"),
-      )
-    }
+    givenDocumentApiRespondsWith(status = responseStatusCode, responseBody = errorResponse)
+
+    val exception = assertThrows<IllegalStateException> { uploadDocument() }
+
+    assertEquals(
+      "Error during uploading document (UUID=%s, StatusCode=%d, Response=%s)".format(
+        "e2487a03-7cf9-4a9c-85e4-1d51efd7b3f1",
+        responseStatusCode,
+        errorResponse,
+      ),
+      exception.message,
+    )
+  }
+
+  @Test
+  fun `Throws an exception when the document is not successful and no response could be parsed`() {
+    givenDocumentApiRespondsWith(status = 401, responseBody = "")
+
+    val exception = assertThrows<IllegalStateException> { uploadDocument() }
```

Comment:
equivalent for assertJ is [here](https://github.com/assertj/assertj-examples/blob/72c339f0f72fd4a6e903c25af2d3f093702f3e4c/assertions-examples/src/test/java/org/assertj/examples/ExceptionAssertionsExamples.java#L268)